### ISSUE TITLE
I have a question about labeling method for classification.

### ISSUE THREAD

**Issue by star4s** [ultralytics/yolov5#10432]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,
If there are several classes on one training image, how should we label them in this case? Normally, if there is one class on one image, we know that the images are put in a folder with the class name.
how should we label multi-class in one training image? 
Thank you for your help.

**Comment by AyushExel** [comment]
@star4s yolov5 currently only supports multi-class classification and not multi-label classification. The labeling structure for mutli-label classification is quite different. The labels are generally in form of a csv file in which each image is assigned the labels in one hot encoded format. 
See this post for example - https://debuggercafe.com/multi-label-image-classification-with-pytorch-and-deep-learning/

@glenn-jocher mutli-label classification is actually more useful than vanilla classification and also has real-world applications. This is something we should add in the long term roadmap

**Comment by glenn-jocher** [comment]
@AyushExel ok got it. We can work on this as a second Classification dataloader.

**Comment by danialmecja** [comment]
@AyushExel Hi, so just to confirm, its technically not possible to carry out multi-label classification when training a YOLOv5 model? My use-case simply is to predict grocery item images where for example for "chips", they have a parent label of "snacks" and its sub-label is the name of its class "chips". I'm guessing from your comment I can't achieve this?

Also, thanks for the link on multi-label classification! However, I'm wondering if its possible to fuse that work around of assigning labels in one-hot encoded format onto a YOLOv5 model as I require a live feed when carrying out predictions. Any advice is greatly appreciated, thanks!

**Comment by AyushExel** [comment]
@danmecj I haven't thought deeply about this but I think yes you can probably just pre-process the dataset to make it compatible with the current training script. One hot encoding might work too but you'll need to write some custom code

**Comment by danialmecja** [comment]
Got it. When you mention pre-processing the dataset, I assume you're referencing the annotation of labels of the images in the YOLOv5 format? And how they are structured in a way where for example: 

1 0.5 0.5 0.2 0.2

Where the left most is referencing the class of the image and the right most values are the bounding box coordinates right? 

I was thinking of adding more classes to the .yaml class list and to add values that reference indexes of the list so it'd look like:

8:1 0.5 0.5 0.2 0.2

Where 8 = "snacks" in the .yaml classes list and 1 = "chips". 

I'll definitely have a look into the training script to see if this kinda format is allowed for labels.

Apologies for the long-winded comment, just seeking clarification as I'm new to model training with YOLO.

**Comment by AyushExel** [comment]
@danmecj no that's the detection dataset format. For classification, yolov5 just accepts the folder-name of the dataset and uses pytorch's ImageDataset class to generate dataset. This is where you might need to add your own dataset handler
